nifi-framework-api/src/main/java/org/apache/nifi/controller/queue/IllegalClusterStateException.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.nifi.controller.queue;
+
```

Comment:
Sorry if this is stupid, but this seems more like an IllegalClusterOperation.   The exception is thrown when you attempt an operation, not when you enter an illegal state.

Also, there is no knowing the "state" you are in.

Reply:
The Exception denotes that the operation cannot performed when the cluster is in the given state. It has the same semantics, in that way, as IllegalStateException - I.e., IllegalStateException does not imply that the object is in an illegal state, but rather that the operation you're trying to perform cannot be performed in the current state. This is analogous. The operation cannot be performed at the current time due to the state of the cluster.

There is no need to know, programmatically what state the cluster is in. The message provided to the Exception can provide information for users to understand the context.